From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: crackerjacques <jack@supremeoverlordjabs.co>
Date: Wed, 29 Jul 2026 19:19:21 +0200
Subject: [ARCHEOLOGY] rockchip64-7.2: carry Anbernic RG Vita Pro patches to
 bleedingedge

> X-Git-Archeology: > recovered message: > Review feedback: also target v7.2 so the board and the rk3576 fixes are
> X-Git-Archeology: > recovered message: > not lost when 7.2 becomes edge. Same 14 patches as rockchip64-7.1 plus
> X-Git-Archeology: > recovered message: > the board DTS; verified to apply against current mainline (7.2-rc):
> X-Git-Archeology: > recovered message: > - 12 apply unchanged
> X-Git-Archeology: > recovered message: > - general-power-supply-add-sgm41542: Kconfig/Makefile contexts
> X-Git-Archeology: > recovered message: > regenerated (new CHARGER_S2M entries landed next to the anchor)
> X-Git-Archeology: > recovered message: > - general-usb-typec-rt1711h-add-husb311: regenerated for the new
> X-Git-Archeology: > recovered message: > designated-initializer style of the rt1711h ID tables
> X-Git-Archeology: > recovered message: > The board DTS needs no changes for 7.2 (all referenced dtsi/pinctrl
> X-Git-Archeology: > recovered message: > labels still exist, no node collisions). The kernel config additions
> X-Git-Archeology: > recovered message: > mirror linux-rockchip64-edge.config.
> X-Git-Archeology: > recovered message: > Signed-off-by: crackerjacques <jack@supremeoverlordjabs.co>
> X-Git-Archeology: - Revision 1845d2b3d9259e6d6aad7dda325ffa82b9087e2e: https://github.com/armbian/build/commit/1845d2b3d9259e6d6aad7dda325ffa82b9087e2e
> X-Git-Archeology:   Date: Wed, 29 Jul 2026 19:19:21 +0200
> X-Git-Archeology:   From: crackerjacques <jack@supremeoverlordjabs.co>
> X-Git-Archeology:   Subject: rockchip64-7.2: carry Anbernic RG Vita Pro patches to bleedingedge
> X-Git-Archeology:
> X-Git-Archeology: - Revision 9105fe7e160b72a7baddc2b005ffff5ae642b376: https://github.com/armbian/build/commit/9105fe7e160b72a7baddc2b005ffff5ae642b376
> X-Git-Archeology:   Date: Wed, 29 Jul 2026 19:19:21 +0200
> X-Git-Archeology:   From: crackerjacques <jack@supremeoverlordjabs.co>
> X-Git-Archeology:   Subject: rockchip64-7.1: rename Vita patches to match patch naming conventions
> X-Git-Archeology:
> X-Git-Archeology: - Revision e67caa6ddb85f5a6191fdfb3d462c10c3ba5cee8: https://github.com/armbian/build/commit/e67caa6ddb85f5a6191fdfb3d462c10c3ba5cee8
> X-Git-Archeology:   Date: Wed, 29 Jul 2026 19:19:21 +0200
> X-Git-Archeology:   From: crackerjacques <jack@supremeoverlordjabs.co>
> X-Git-Archeology:   Subject: rockchip64-edge: device drivers for Anbernic RG Vita Pro
> X-Git-Archeology:
---
 drivers/usb/typec/tcpm/tcpci_rt1711h.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/usb/typec/tcpm/tcpci_rt1711h.c b/drivers/usb/typec/tcpm/tcpci_rt1711h.c
index 111111111111..222222222222 100644
--- a/drivers/usb/typec/tcpm/tcpci_rt1711h.c
+++ b/drivers/usb/typec/tcpm/tcpci_rt1711h.c
@@ -377,6 +377,7 @@ static const struct rt1711h_chip_info rt1715 = {
 
 static const struct i2c_device_id rt1711h_id[] = {
 	{ .name = "et7304", .driver_data = (kernel_ulong_t)&rt1715 },
+	{ .name = "husb311", .driver_data = (kernel_ulong_t)&rt1711h },
 	{ .name = "rt1711h", .driver_data = (kernel_ulong_t)&rt1711h },
 	{ .name = "rt1715", .driver_data = (kernel_ulong_t)&rt1715 },
 	{ }
@@ -385,6 +386,7 @@ MODULE_DEVICE_TABLE(i2c, rt1711h_id);
 
 static const struct of_device_id rt1711h_of_match[] = {
 	{ .compatible = "etekmicro,et7304", .data = &rt1715 },
+	{ .compatible = "hynetek,husb311", .data = &rt1711h },
 	{ .compatible = "richtek,rt1711h", .data = &rt1711h },
 	{ .compatible = "richtek,rt1715", .data = &rt1715 },
 	{}
-- 
Armbian

